src/core/Akka/IO/SocketEventArgsPool.cs

```diff
@@ -116,5 +118,27 @@ public static void SetBuffer(this SocketAsyncEventArgs args, ByteString data)
                 }
             }
         }
+        
+        public static void SetBuffer(this SocketAsyncEventArgs args, IEnumerable<ByteString> dataCollection)
```

Comment:
Any chance that these changes might help address https://github.com/akkadotnet/akka.net/issues/3921 @IgorFedchenko ?

Reply:
I do not think so. My update here is related to `SocketAsyncEventArgs` buffers initialization before sending data to socket. And in referenced issue they are not writing, but reading corrupted messages (also, since it is opened for 24 days since last commit, I guess they do not reproduce issue anymore). So, no, this one should not be related.